### ISSUE TITLE
==/!= operators on structs and comparing to null

### ISSUE THREAD

**Issue by kindermannhubert** [dotnet/roslyn#23178]
**Version Used**: VS 15.4.4

**Steps to Reproduce**:
This code:
```C#
struct MyStruct
{
}

static void Foo(MyStruct myStruct)
{
    if (myStruct == null) myStruct.ToString();
}
```
does not compile with error "Operator '==' cannot be applied to operands of type 'MyStruct' and '\<null\>'.". That's ok.

But code:
```C#
struct MyStruct
{
    public static bool operator !=(MyStruct a, MyStruct b) => false;
    public static bool operator ==(MyStruct a, MyStruct b) => true;
}

static void Foo(MyStruct myStruct)
{
    if (myStruct == null) myStruct.ToString();
}
```
compiles fine. After disassembling you can see no code for "if (myStruct == null) myStruct.ToString();" is emitted (which is correct).

**Expected Behavior**:
I believe second example should not compile as well.

**Comment by HaloFour** [comment]
This is expected behavior.  [Per Eric Lippert](https://stackoverflow.com/questions/2022425/comparing-structs-to-null/2022553#2022553):

> The choice is actually between `(obj, obj), (Foo, Foo) and (Foo?, Foo?)` -- defining an equality operator on `(Foo, Foo)` automatically also defines the lifted-to-nullable version.

**Comment by kindermannhubert** [comment]
Thanks, I understand.

But still I think it should be reported by compiler.

At the same link you provided:
> Nope, we didn't miss this one. (Though the fact that a warning is not reported seems to be a bug.)  This behaviour is correct according to the specification.                    -Eric Lippert

**Comment by CyrusNajmabadi** [comment]
>But still I think it should be reported by compiler.

This was tried at one point, but it broke many customers, violating our rules on compat.

We have a concept called "warning waves" where customers could opt into more aggressive warnings.  This warning is known to be a prime example of one we'd like to add once that system is built.

**Comment by jcouv** [comment]
Linking to warning waves: https://github.com/dotnet/roslyn/issues/1580

**Comment by Youssef1313** [comment]
Duplicate of https://github.com/dotnet/roslyn/issues/30196

**Comment by f-i-x-7** [comment]
Hi all, I've just recently browsed across 'New Feature - Warning Waves' issues and found out this issue.

> Duplicate of #30196

As this is duplicate, it should be closed. CS8073 is triggered in sampe code, work is done.
